pkg/engine/task/task_apply.go

```diff
@@ -135,9 +138,12 @@ func applyResources(rr []runtime.Object, ctx Context) ([]runtime.Object, error)
 		case err != nil: // raise any error other than StatusReasonNotFound
 			return nil, err
 		default: // update existing resource
-			err := patchResource(r, existing, ctx)
-			if err != nil {
-				return nil, fmt.Errorf("failed to patch %s/%s: %w", key.Namespace, key.Name, err)
+			// skip updating v1.Job(s): it is not possible to update the Job.Spec after the job was created
+			if r.GetObjectKind().GroupVersionKind() != jobGVK {
```

Comment:
what if the update was actually updating something essential (like a URL to something) we just *silently* skip the update here? 🤔maybe at least add a log line?

 is it true that for every job that is part of some operator it has to be deleted after it was run with KUDO or you'll never be able to patch it? Should we validate against that somehow as well?

Reply:
Well, I think that's not really an issue with KUDO, that applies to Jobs in general.

I agree with the logging - If we're going to add this in, we should at least log something

Reply:
looking at the issue https://github.com/kudobuilder/kudo/issues/1425#issue-580829626

if the job spec contains "${ISTIO_GATEWAY_HOSTNAME}" will that ever be possible to update it? I don't think so - so should jobs always be deleted after finished or should there never be templated variables inside?

Reply:
Well, `${ISTIO_GATEWAY_HOSTNAME}` is a bash variable that's set inside the script - So that shouldn't be an issue. But apart from that, yeah - I don't think the job spec template can be updated

Reply:
> on a second (or rather third, fourth) thought, I am not sure this is right. I think there might be job with parameters outside spec and we should allow that update. Maybe the right solution is to rather handle that error than not doing the update?

`Job.spec` is effectively immutable and can't be updated 🤷 So a job with parameters is effectively a no-go unless we delete-recreate them.

Reply:
yeah you can still theoretically use them to update metadata or something? 🤔

Reply:
you mean to update `job.spec.metadata`? No, it is also immutable and can not be changed.